src/preload.ts

```diff
@@ -8,6 +8,45 @@ import {
   PrinterInfo,
   channel as getPrinterInfoChannel,
 } from './ipc/get-printer-info'
+import { channel as getDeviceListChannel } from './ipc/get-device-list'
+import { Device } from 'usb-detection'
+import {
+  channel as subscribeToDevicesChannel,
+  UpdateType,
+  deviceUpdateChannel,
+} from './ipc/subscribe-to-devices'
+
+class Listeners<A extends any[], C extends Function = (...args: A) => void> {
```

Comment:
feels a bit like overkill to build in ability for multiple listeners? I can live with it, just wondering if this complexity is needed.

Reply:
Okay, changed it to a single listener. Note that I made it more robust in terms of actually turning monitoring off when nobody is listening, so it's already got a fair amount of inherent complexity.